src/packages/utils/tests/json-rpc-data.test.ts

```diff
@@ -0,0 +1,187 @@
+      validBytelengths.forEach(bytelength => {
+        const d = new Data("0x01", bytelength);
+      });
+    });
+
+    it("should fail with invalid bytelengths", () => {
+      invalidBytelengths.forEach(bytelength => {
+        assert.throws(() => {
+          const d = new Data("0x01", bytelength);
+        }, undefined, `Should fail to accept bytelength: ${bytelength} of type: ${typeof bytelength}`);
+      });
+    });
+  });
+
+  describe("from()", () => {
+    it("should accept different representations of value", () => {
+      validValues.forEach(value => {
+        const d = Data.from(value);
+      });
+    });
+
+    it("should fail with invalid values", () => {
+      invalidValues.forEach(value => {
+        assert.throws(() => {
+          const d = Data.from(value);
+        }, undefined, `Should fail to accept value: ${value} of type: ${typeof value}`);
+      });
+    });
+
+    it("should accept valid bytelengths", () => {
+      validBytelengths.forEach(bytelength => {
+        const d = Data.from("0x01", bytelength);
+      });
+    });
+
+    it("should fail with invalid bytelengths", () => {
+      invalidBytelengths.forEach(bytelength => {
+        assert.throws(() => {
+          const d = Data.from("0x01", bytelength);
+        }, undefined, `Should fail to accept bytelength: ${bytelength} of type: ${typeof bytelength}`);
+      });
+    });
+  });
+
+  describe("toString()", () => {
+    it("should stringify without arguments", () => {
+      validValues.forEach(value => {
+        const d = new Data(value);
+        const s = d.toString();
+        const expected = getExpectedString(value);
+
+        assert.equal(s, expected);
+      });
+    });
+
+    it("should stringify with valid bytelengths", () => {
+      validValues.forEach(value => {
+        const d = new Data(value);
+        validBytelengths.forEach(bytelength => {
+          const s = d.toString(bytelength);
+          const expected = getExpectedString(s, bytelength);
+
+          assert.equal(s, expected);
+        });
+      });
+    });
+
+    it("should fail with invalid bytelengths", () => {
+      validValues.forEach(value => {
+        const d = new Data(value);
+        invalidBytelengths.forEach(bytelength => {
+          assert.throws(() => {
+            const s = d.toString(<any>bytelength);
+          }, undefined, `Should fail to accept bytelength: ${bytelength} of type: ${typeof bytelength}`);
+        });
+      });
+    });
+
+    it("should pad the value with larger bytelength", () => {
+      validValues.forEach(value => {
+        const d = new Data(value);
+        const s = d.toString(10);
+        const expected = getExpectedString(s, 10);
+
+        assert.equal(s, expected);
+      });
+    });
+
+    it("should truncate the value with a smaller bytelength", () => {
+      const d = new Data("0x1234567890abcdef");
+      const s = d.toString(2);
+
+      assert.equal(s, "0x1234");
```

Comment:
Can we use your `getExpectedString` function here too just for consistency? That function can truncate too, right?